config/v1/types_authentication.go

```diff
@@ -717,37 +754,55 @@ type PrefixedClaimMapping struct {
 	Prefix string `json:"prefix"`
 }
 
-// TokenValidationRuleType represents the different
-// claim validation rule types that can be configured.
+// TokenValidationRuleType defines the type of token validation rule.
 // +enum
+// +openshift:validation:FeatureGateAwareEnum:featureGate="",enum="RequiredClaim";
```

Comment:
This isn't persisting correctly, I think it is because we are missing an equivalent marker for the `ExternalOIDC` and `ExternalOIDCWithUIDAndExtraClaimMappings` feature gates.

Adding:
```go
// +openshift:validation:FeatureGateAwareEnum:featureGate=ExternalOIDC,enum="RequiredClaim";
// +openshift:validation:FeatureGateAwareEnum:featureGate=ExternalOIDCWithUIDAndExtraClaimMappings,enum="RequiredClaim";
```

Should fix that.